### PR TITLE
[concurrency] Add logs, tests, and comments

### PR DESCRIPTION
**Pull request by lbernick**
This commit adds more logs for the concurrency controller, and exports + tests the Match function.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- n/a Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Commit messages includes a project tag in the subject - e.g. [OCI], [hub], [results], [task-loops]

_See [the contribution guide](https://github.com/tektoncd/experimental/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: concurrency/pkg/reconciler/concurrency/concurrencycontrol.go [hunk 1]
```diff
@@ -127,11 +133,17 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, pr *v1beta1.PipelineRun)
 	return r.updateLabelsAndStartPipelineRun(ctx, pr)
 }
 
-func matches(pr *v1beta1.PipelineRun, cc *v1alpha1.ConcurrencyControl) bool {
+// Matches returns true if the PipelineRun is selected by the ConcurrencyControl's selector.
+// An empty selector always matches the PipelineRun.
+func Matches(pr *v1beta1.PipelineRun, cc *v1alpha1.ConcurrencyControl) bool {
```
Comment (jerop) [inline concurrency/pkg/reconciler/concurrency/concurrencycontrol.go:138]: why do you export `Matches`?

Comment (lbernick) [reply concurrency/pkg/reconciler/concurrency/concurrencycontrol.go:138]: In this PR, primarily to make it easier to test. However, I think it's useful to know whether a concurrency control applies to a given pipelinerun (i.e. whether the control matches the pipelinerun) and would be a useful function to export.

**Pull request merged**